app/src/main/java/de/tum/in/tumcampusapp/component/ui/cafeteria/details/CafeteriaDetailsSectionFragment.kt

```diff
@@ -63,9 +64,14 @@ class CafeteriaDetailsSectionFragment : Fragment() {
         with(binding) {
             menuDateTextView.text = menuDateString
 
-            val hours = OpenHoursHelper(requireContext()).getHoursByIdAsString(cafeteriaId, menuDate)
-            menuOpeningHours.text = hours
-            menuOpeningHours.isVisible = hours.isNotEmpty()
+            // Update the left time for opening/closing every one minute
+            timer.schedule(object : TimerTask(){
+                override fun run(){
+                    val hours = OpenHoursHelper(requireContext()).getHoursByIdAsString(cafeteriaId, menuDate)
```

Comment:
@Bentipa @joschahenningsen what do you think would be the best option? The number of methods, that have a context guarantee, is weird, and I don't know which one to choose…

Reply:
I have added a potential fix for the problem. It is not ideal, but solves the issue of crashing the entire app. Instead of requireContext() the context is retrieved from the view element, where it should exist as far as I understand is. To be sure, I added the null check.